### PR TITLE
plaid oauth handling - wallet part

### LINKED ISSUE #1444: [M2] OAuth Setup
> OAuth support is required in all Plaid integrations. Without OAuth support, your end users will not be able to connect accounts from institutions that require OAuth such as Chase and Capital One

Documentation: https://plaid.com/docs/link/oauth/

AC:
* [Request production access from Plaid](https://plaid.com/docs/link/oauth/#request-production-access-from-plaid)
* [Complete Registration requirements](https://plaid.com/docs/link/oauth/#complete-the-registration-requirements)
* Double check that we have a redirect URI created for the app
* Add redirect_uri for iOS and android_package_name for Android on the link/token/create call [in-house-liquidity code](https://github.com/valora-inc/in-house-liquidity/blob/0ccf69b707a8c8fc18b0f07f2f6bfb87691923e6/src/clients/plaid-client-wrapper.ts#L101-L103)
* [Verify the Oauth setup in the sandbox environment](https://plaid.com/docs/link/oauth/#testing-oauth)

### PR DESCRIPTION
**Pull request by lisaqyli**
### Description
This PR handles deeplink from plaid linking bank account success flow. After user successfully linked their bank account through plaid, some of the bank accounts requires user to go through OAuth flow. https://valoraapp.com/plaid-oauth-redirect is provided as the link to redirect to after oauth success. 

After the deeplink is opened from the app, we will navigate user to bankAccounts screen

### Tested

pending e2e test

### Related issues

- Fixes #1444

### CODE REVIEW

File: packages/mobile/src/utils/linking.ts [hunk 1]
```diff
@@ -13,8 +15,12 @@ export function navigateToWalletStorePage() {
...
 }
 
-export function isDeepLink(url: string): boolean {
-  return url.startsWith('celo:')
+export function isDeepLink(deeplink: string): boolean {
+  return deeplink.startsWith('celo:')
+}
+
+export function isPlaidRedirectDeepLink(deepLink: string): boolean {
+  return decodeURIComponent(deepLink).startsWith(PLAID_UNIVERSAL_LINK_PREFIX)
```
Comment (jeanregisser) [inline packages/mobile/src/utils/linking.ts:23]: Is `decodeURIComponent` necessary?

**Review by jeanregisser — COMMENTED** [review]
Great!

Is this oauth interaction happening with an In-App browser or in the user browser?
I'm asking because universal links on iOS can [sometimes fail](http://www.openradar.me/radar?id=4999496467480576). 
This is currently affecting WalletConnect universal links on iOS for instance.

So if this is flow is always started or happening within the app, it would be more reliable to use the `celo://` urls.

Happy to chat about it.

File: packages/mobile/src/utils/linking.ts (hunk 1 above)
Comment (lisaqyli) [reply packages/mobile/src/utils/linking.ts:23]: So the reason why using a universal link instead of `celo://` is Plaid requires us to use universal link, See the react native (iOS) section here https://plaid.com/docs/link/oauth/#create-and-register-a-redirect-uri

I'm actually not sure if the decodeURIComponent is necessary, I'll try to test e2e to see if it's something i need to fix.,

File: packages/mobile/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -33,9 +33,11 @@
         <action android:name="android.intent.action.VIEW" />
         <category android:name="android.intent.category.DEFAULT" />
         <category android:name="android.intent.category.BROWSABLE" />
-        <data android:scheme="https" android:host="valoraapp.com" android:pathPrefix="/wc" />
+        <data android:scheme="https" android:host="valoraapp.com" />
+        <data android:pathPrefix="/wc" />
```
Comment (cajubelt) [inline packages/mobile/android/app/src/main/AndroidManifest.xml:37]: can you do some basic manual test to make sure this doesn't break WC on Android?

File: packages/mobile/ios/Podfile.lock [hunk 3]
```diff
@@ -9,7 +9,7 @@ PODS:
     - SDWebImage (~> 5.1)
   - clevertap-react-native (0.5.2):
     - CleverTap-iOS-SDK (= 3.9.3)
-    - React-Core
```
Comment (cajubelt) [inline packages/mobile/ios/Podfile.lock:12]: looks like a lot of this is unrelated, can you split it out into a separate PR?

**Review by cajubelt — COMMENTED** [review]
nice!

**Review by jh2oman — APPROVED** [review]

**Review by cajubelt — APPROVED** [review]
🥳

**Comment by ValoraQA** [comment]
Hey @MuckT , Please let us know do we need to perform testing from our side.

**Comment by MuckT** [comment]
@lisaqyli could you let QA know how they should test if needed? Thanks in advance!

**Comment by ValoraQA** [comment]
@lisaqyli please let us know whether it needs to be tested or not .
Thanks..!!
